open_humans/account_views.py

```diff
@@ -141,3 +174,149 @@ def delete(self, request, *args, **kwargs):
+        login_url = passthrough_next_redirect_url(self.request,
+                                                  reverse("account_login"),
+                                                  'next')
+        # NOTE: For backwards compatibility
+        ret['password_reset_form'] = ret.get('form')
+        # (end NOTE)
+        ret.update({"login_url": login_url})
+        return ret
+
+    def post(self, request, *args, **kwargs):
+        request = super().post(request, *args, **kwargs)
+        return request
+
+
+class PasswordResetFromKeyView(FormView):
+    """
+    Let's get a new password!
+    """
+    template_name = 'account/password_reset_token.html'
+    form_class = PasswordResetForm
+
+    def _get_user(self, uidb36):
+        User = get_user_model()
+        try:
+            pk = url_str_to_user_pk(uidb36)
+            return User.objects.get(pk=pk)
+        except (ValueError, User.DoesNotExist):
+            return None
+
+    def dispatch(self, request, uidb36, key, **kwargs):
+        self.request = request
+        self.key = key
+        user = self._get_user(uidb36)
+        self.reset_user = user
+        if user is None:
+            return redirect('account-password-reset-fail')
+        token = default_token_generator.check_token(user, key)
+        if not token:
+            return redirect('account-password-reset-fail')
+        return super().dispatch(request, uidb36, key, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        ret = super().get_context_data(**kwargs)
+        ret['action_url'] = reverse(
+            'account_reset_password_from_key',
+            kwargs={'uidb36': self.kwargs['uidb36'],
+                    'key': self.kwargs['key']})
+        return ret
+
+    def change_password(self, password):
+        user = self.reset_user
+        user.set_password(password)
+        user.save()
+        return user
+
+    def form_valid(self, form):
+        if form.is_valid():
+            self.change_password(form.clean_password())
+
+            perform_login(
+                self.request, self.reset_user,
+                email_verification=EMAIL_VERIFICATION)
+            member = Member.objects.get(user=self.reset_user)
+            next_url = member.password_reset_redirect
+            member.password_reset_redirect = '' # Clear redirect from db
+            member.save()
+            messages = {
+                'settings_updated': {
+                    'level': django_messages.SUCCESS,
+                    'text': 'Password successfully reset.'},}
+            return redirect(unquote_plus(next_url))
+
+
+class PasswordChangeView(PasswordChangeView):
+    """
+    Change the password
+    """
+    template_name = 'account/password_change.html'
+    form_class = ChangePasswordForm
+    success_url = reverse_lazy('my-member-settings')
+    def form_valid(self, form):
+
+        return super().form_valid(form)
+
+
+class ConfirmEmailView(ConfirmEmailView):
```

Comment:
(As mentioned elsewhere, should avoid repeating the parent class name.)

Reply:
Fixed